tests/backup/backup_helper.go

```diff
@@ -5559,3 +5679,175 @@ func UpgradeKubevirt(versionToUpgrade string, workloadUpgrade bool) error {
 	}
 	return nil
 }
+
+// ChangeAdminNamespace changes admin namespace for Storage Cluster
+func ChangeAdminNamespace(namespace string) (*v1.StorageCluster, error) {
+	// Get current storage cluster configuration
+	isOpBased, _ := Inst().V.IsOperatorBasedInstall()
+	storkDeploymentNamespace, err := k8sutils.GetStorkPodNamespace()
+	if err != nil {
+		return nil, err
+	}
+	stc, err := Inst().V.GetDriver()
+	if err != nil {
+		return nil, err
+	}
+	log.Infof("Is op based deployment %v", isOpBased)
+
+	if isOpBased {
+		if adminNamespace, ok := stc.Spec.Stork.Args["admin-namespace"]; ok {
+			log.Info("Current admin namespace - [%s]", adminNamespace)
+		}
+		// Setting the new admin namespace
+		if namespace != "" {
+			stc.Spec.Stork.Args["admin-namespace"] = namespace
+		} else {
+			delete(stc.Spec.Stork.Args, "admin-namespace")
+		}
+		stc, err = operator.Instance().UpdateStorageCluster(stc)
+		if err != nil {
+			return nil, err
+		}
+		if namespace != "" {
+			log.Infof("Configured admin namespace to %s", namespace)
+		} else {
+			log.Infof("Removed admin namespace") // Removing admin namespace is not supported - https://docs.portworx.com/portworx-backup-on-prem/configure/admin-namespace.html
+		}
+	} else {
+		log.Infof("Updating stork deployment as it's pxe is not present")
+		storkDeployment, err := apps.Instance().GetDeployment(storkDeploymentName, storkDeploymentNamespace)
+		if err != nil {
+			return nil, err
+		}
+		storkDeployment.Spec.Template.Spec.Containers[0].Command = append(storkDeployment.Spec.Template.Spec.Containers[0].Command, fmt.Sprintf("--admin-namespace=%s", namespace))
+
+		_, err = apps.Instance().UpdateDeployment(storkDeployment)
+		if err != nil {
+			return nil, err
+		}
+	}
+	time.Sleep(5 * time.Minute)
```

Comment:
Can we fetch the pod status and sleep instead of blind 5 min sleep?

Reply:
Why not run it as a function with something like the below code snippet: 
```
	storkStatusCheck := func() (interface{}, bool, error) {
               your code goes here 
		if err != nil {
			return "", true, err
		}
		return "", false, nil
	}
	_, err = task.DoRetryWithTimeout(storkStatusCheck, 1*time.Minute, 10*time.Second)
```

Reply:
We are validating stork deployment only at line 5735:
`err = apps.Instance().ValidateDeployment(updatedStorkDeployment, storkPodReadyTimeout, podReadyRetryTime)`